db/migrate/20190925175757_add_polymorphic_reference_to_icons.rb

```diff
@@ -0,0 +1,19 @@
+class AddPolymorphicReferenceToIcons < ActiveRecord::Migration[5.2]
+  def up
+    add_reference :icons, :iconable, :polymorphic => true
+
+    Icon.all.each do |icon|
+      icon.update(
+        :iconable_type => "PortfolioItem",
+        :iconable_id   => icon.portfolio_item_id
+      )
+    end
+
+    remove_column :icons, :portfolio_item_id, :bigint
+  end
+
+  def down
+    add_column :icons, :portfolio_item_id, :bigint
+    remove_reference :icons, :iconable, :polymorphic => true
```

Comment:
Is there value in moving the data back from the polymorphic columns into the `:portfolio_item_id`? We've had some migration issues in the past, so the chances for losing data could happen if we need to down the migration.